### ISSUE TITLE
[HOLD for payment 2023-04-05] [$1000] No placeholder text and 'no result found' message is also not present on no result on new pronouns search page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15779]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app in any of this devices: android chrome, mac safari, mac chrome, windows chrome
2. Open settings
3. Open profile
4. Open Pronouns
5. Observe search field for placeholder
6. Search term which does not exists eg: Test

## Expected Result:
App should display placeholder text in search and should display 'No result found' message when no result exists as it does on all other search pages throughout the app OR if want to make it similar to timezone search page, it should have default selected value already searched in the search box

## Actual Result:
App does not have any placeholder text or 'No result found' message for pronouns page

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.81-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/224105851-7fcf5055-b108-4f52-8c71-33a614c92378.mp4

https://user-images.githubusercontent.com/43996225/224105870-e28ac1fe-d4d7-482c-a81d-30d9e7fedf35.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1678093363937389

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01800cb0ea689d7c0e</li>
        <li>Upwork Job ID: 1638901127704383488</li>
        <li>Last Price Increase: 2023-03-23</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @muttmuure (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@muttmuure Eep! 4 days overdue now. Issues have feelings too...

**Comment by MelvinBot** [comment]
@muttmuure 6 days overdue. This is scarier than being forced to listen to Vogon poetry!

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
No placeholder text and 'no result found' message is also not present on no result on new pronouns search page

### What is the root cause of that problem?
Within `PronounsPage.js` we can see that `placeholderText` and `headerMessage` prop not set for `<OptionsSelector>` component. So it will not show placeholder and no results found message at present. This is the root cause of the problem.

https://github.com/Expensify/App/blob/acc4984ce41d21e0f89c1913053232a567598d8c/src/pages/settings/Profile/PronounsPage.js#L123-L133

### What changes do you think we should make in order to solve the problem?
We have to prepare `headerMessage` based on the `filteredPronounsList` array i.e. If search text input have some value and `filteredPronounsList` empty then show no results found message. 

For placeholder we can set one pronoun as placeholder text. Or We can create new key and set for placeholderText props.

Within `src/pages/settings/Profile/PronounsPage.js` file we have to update code as shown below:

```javascript
render() {
 const filteredPronounsList = this.getFilteredPronouns();
 const headerMessage = this.state.searchValue.trim() && !filteredPronounsList.length ? this.props.translate('common.noResultsFound') : '';  // *** ADD THIS ***

  <OptionsSelector
      textInputLabel={this.props.translate('pronounsPage.pronouns')}
      placeholderText={this.props.translate('pronouns.heHimHis')}   // *** ADD THIS ***
      ...
      onChangeText={this.onChangeText}
      headerMessage={headerMessage}   // *** ADD THIS ***
      ...
  />
```
So this will solve both problem i.e. Set placeholder text and No results found message.

### What alternative solutions did you explore? (Optional)
None

**Comment by PankajAS** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Pronouns search list not showing without search and 'no result found' not showing when there is no data in list.

### What is the root cause of that problem?
We are not sending  `headerMessage` prop value with `OptionsSelector` component in `PronounsPage.js`, `getFilteredPronouns` return empty list when search input doesn't any value which is wrong condition.

### What changes do you think we should make in order to solve the problem?
First we have to show list of pronounces which is not showing without search then we have to send  `headerMessage={headerMessage}` with `OptionsSelector` based on length of `filteredPronounsList` data.

### What alternative solutions did you explore? (Optional)
None

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01800cb0ea689d7c0e

**Comment by puneetlath** [comment]
@muttmuure I went ahead and marked this external for you. Hope you don't mind.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @sobitneupane (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jasperhuangg (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by dhairyasenjaliya** [comment]
### Proposal

**Please re-state the problem that we are trying to solve in this issue.**

- No placeholder text and 'no result found' message is also not present on no result on new pronouns search page

**What is the root cause of that problem?**

- Currently, we do not have any validation to check if searched values have empty data or not based on that we need to show no result found, we already `headerMessage`  props to display a message we just need to define logic whether the user has searched value and if yes then need to count the length of `filteredPronounsList`  

**What changes do you think we should make in order to solve the problem?**

- To define logic and make sure the user has inputted search value and check the length of `filteredPronounsList`  

- Pass the `headerMessage` props to `<OptionsSelector>`

```diff
+     const headerMessage = this.state.searchValue.trim() && !filteredPronounsList.length ? this.props.translate('common.noResultsFound') : '';

                            <OptionsSelector
                                textInputLabel={this.props.translate('pronounsPage.pronouns')}
                                sections={[{data: filteredPronounsList}]}
                                value={this.state.searchValue}
+                                headerMessage={headerMessage}
                                onSelectRow={this.updatePronouns}
                                onChangeText={this.onChangeText}
                                optionHoveredStyle={styles.hoveredComponentBG}
                                safeAreaPaddingBottomStyle={safeAreaPaddingBottomStyle}
                                shouldFocusOnSelectRow
                                shouldHaveOptionSeparator
                            />
+
```

**What alternative solutions did you explore:** None

**Result**
<img src=https://user-images.githubusercontent.com/47522946/227219797-84abb45a-4ac5-4acb-aaee-075be9472f07.png  width=50%>

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
No placeholder text and 'no result found' message is also not present on no result on new pronouns search page

### What is the root cause of that problem?
Within `PronounsPage.js` we can see that `placeholderText` and `headerMessage` prop not set for `<OptionsSelector>` component. So it will not show placeholder and no results found message at present. This is the root cause of the problem.

https://github.com/Expensify/App/blob/acc4984ce41d21e0f89c1913053232a567598d8c/src/pages/settings/Profile/PronounsPage.js#L123-L133

### What changes do you think we should make in order to solve the problem?
We have to prepare `headerMessage` based on the `filteredPronounsList` array i.e. If search text input have some value and `filteredPronounsList` empty then show no results found message. 

For placeholder we can set one pronoun as placeholder text. Or We can create new key and set for placeholderText props.

Within `src/pages/settings/Profile/PronounsPage.js` file we have to update code as shown below:

```javascript
render() {
 const filteredPronounsList = this.getFilteredPronouns();
 const headerMessage = this.state.searchValue.trim() && !filteredPronounsList.length ? this.props.translate('common.noResultsFound') : '';  // *** ADD THIS ***

  <OptionsSelector
      textInputLabel={this.props.translate('pronounsPage.pronouns')}
      placeholderText={this.props.translate('pronouns.heHimHis')}   // *** ADD THIS ***
      ...
      onChangeText={this.onChangeText}
      headerMessage={headerMessage}   // *** ADD THIS ***
      ...
  />
```
So this will solve both problem i.e. Set placeholder text and No results found message.

### What alternative solutions did you explore? (Optional)
None

**Comment by PankajAS** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Pronouns search list not showing without search and 'no result found' not showing when there is no data in list.

### What is the root cause of that problem?
We are not sending  `headerMessage` prop value with `OptionsSelector` component in `PronounsPage.js`, `getFilteredPronouns` return empty list when search input doesn't any value which is wrong condition.

### What changes do you think we should make in order to solve the problem?
First we have to show list of pronounces which is not showing without search then we have to send  `headerMessage={headerMessage}` with `OptionsSelector` based on length of `filteredPronounsList` data.

### What alternative solutions did you explore? (Optional)
None

**Comment by MelvinBot** [comment]
@sobitneupane @jasperhuangg @muttmuure this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by sobitneupane** [comment]
As suggested by @PankajAS,

> First we have to show list of pronounces which is not showing without search

Similar to search result, we should show the list of pronouns if searchvalue is empty. 

But looking at the code, it looks like we are intentionally showing nothing when searchvalue is empty.

https://github.com/Expensify/App/blob/b0b31d1b9f424a443a5a475a37ba2029c31a7dbf/src/pages/settings/Profile/PronounsPage.js#L65-L72

Even when an pronoun is already selected it only shows up after searching.

https://user-images.githubusercontent.com/25876548/227423065-77f56301-7212-4341-b642-9ed14e9a1d4f.mov

>
>

@jasperhuangg @puneetlath(As he reviewed the [issue](https://github.com/Expensify/App/issues/15211) to add search in pronouns page )
Should we show the list of pronouns if search value is empty?

**Comment by dhairyasenjaliya** [comment]
@sobitneupane I think intentionally to hide all pronouns by default and the other case will handle here for showing default selection https://github.com/Expensify/App/issues/16349#issuecomment-1481288035

**Comment by PrashantMangukiya** [comment]
@sobitneupane yes we are not showing suggestion when search text is empty. Also according to guideline proposal submitted before Help Wanted label should not be considered here is [link](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job).

> Do not propose solutions to jobs until the Help Wanted label has been applied. Any proposals submitted before these labels are added will not be reviewed.

**Comment by sobitneupane** [comment]
Thanks @dhairyasenjaliya @PrashantMangukiya.

Let's hear from @jasperhuangg and @puneetlath about https://github.com/Expensify/App/issues/15779#issuecomment-1482231795 before proceeding.

**Comment by jasperhuangg** [comment]
Thanks @dhairyasenjaliya for the succinct proposal! Let's implement it.

**Comment by MelvinBot** [comment]
📣 @dhairyasenjaliya You have been assigned to this job by @jasperhuangg!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by PrashantMangukiya** [comment]
@jasperhuangg I will request to re-review my proposal. Because as per scope of the issue I proposed solution for both that asked i.e. **"No placeholder text"** and **"no result found message"** As my proposal covers both, so it should be considered. 

Thank you for your time.

**Comment by PrashantMangukiya** [comment]
@sobitneupane I will request to re-review my proposal. Because as per scope of the issue I proposed solution for both that asked i.e. **"No placeholder text"** and **"no result found message"** As my proposal covers both, so it should be considered.  Thank you for your time.

**Comment by PrashantMangukiya** [comment]
@jasperhuangg @sobitneupane I will request to re-review my proposal. Because as per scope of the issue I proposed solution for both that asked i.e. **"No placeholder text"** and **"no result found message"** As my proposal covers both, so it should be considered. 

Thank you for your time.

**Comment by PrashantMangukiya** [comment]
@stitesExpensify Bump https://github.com/Expensify/App/issues/15779#issuecomment-1483618591

**Comment by dhairyasenjaliya** [comment]
Alright thnx @jasperhuangg lets implement ,
@PrashantMangukiya setting placeholder is not scope of this PR and also we are handling placeholder + clear button on separate issue checkout this https://github.com/Expensify/App/issues/16349#issuecomment-1481288035

**Comment by PrashantMangukiya** [comment]
@dhairyasenjaliya That discussion is related to clear button and set selected value  see https://github.com/Expensify/App/issues/16349#issuecomment-1481274917 That discussion is not about placeholder. 

Here we are talking about placeholder. As per **title and expected result** of this issue **Setting placeholder** is in the scope. Also C+ review not done by @sobitneupane  for all proposal posted on this issue. So it should be done before assign the project to anyone.

cc @sobitneupane @jasperhuangg

**Comment by dhairyasenjaliya** [comment]
@PrashantMangukiya we are intentionally hiding placeholder as per pre design only even if we want to add we can ask design team first , but I don't think this will be added still you can try thnx @sobitneupane @jasperhuangg

**Comment by PrashantMangukiya** [comment]
@dhairyasenjaliya either placeholder should be kept or not is a secondary thing. 

Here main point is whatever asked in the issue should be covered by the proposal. During proposal submit time as a contributor we can not decide which thing exclude from the issue scope and submit proposal accordingly etc. 

So at the time of proposal submission we have to consider what is asked in the issue scope and proposal should cover those things with efficient solution. So thereafter if team decide to not include some feature it is  a secondary thing. But It should not change the proposal selection process.

So here my concern is, my proposal includes all that is asked in the issue scope. So it should be considered.

cc @sobitneupane @jasperhuangg

**Comment by dhairyasenjaliya** [comment]
If we need to add a placeholder then I have updated my proposal just because none of the proposals handles language changes https://github.com/Expensify/App/issues/15779#issuecomment-1481234835 (I still believe will not be required here)

cc @sobitneupane @jasperhuangg

**Comment by PrashantMangukiya** [comment]
I already mentioned this things in my original proposal as shown below. So based on the what team decide we have to set as placeholder text. And still my proposal is unedited.

> For placeholder we can set one pronoun as placeholder text. Or We can create new key and set for placeholderText props.

**Comment by dhairyasenjaliya** [comment]
The proposal is not complete didn't mention anything about finding the current key/functions just mentioned adding a new key where else there is no need to define a new key we already have it in the code
for no result, there was a clear explanation but for the placeholder, there is no clear explanation

**Comment by PrashantMangukiya** [comment]
At the time of proposal submit it is not necessary to provide line by line code. General idea is enough what should be implemented where. At the time of pr thereafter team will decide either existing keys used, or create new key or set value from saved pronoun or which is the best approach.

> Do not make extensive changes to your current proposal until after it has been reviewed.

Refer this [guideline](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job:~:text=Do%20not%20make%20extensive%20changes%20to%20your%20current%20proposal%20until%20after%20it%20has%20been%20reviewed.)

**Comment by dhairyasenjaliya** [comment]
Agreed im not saying about line-by-line code as well but general information on how to get the key should be mentioned as it clearly we have it on code based we can't just assume and let the reviewer share how to do it our job to describe and mention how this can be done :)

**Comment by jasperhuangg** [comment]
Apologies guys I messed up and didn't realize that those placeholder changes also weren't in scope.

@dhairyasenjaliya Sorry I assigned you incorrectly. It seems @PrashantMangukiya was correct in that their proposal accounted for everything that was in scope. I know you modified your proposal to include those changes but @PrashantMangukiya was first, so I'm reassigning this issue to them.

Thanks for helping to clear this up everyone.

**Comment by MelvinBot** [comment]
📣 @PrashantMangukiya You have been assigned to this job by @jasperhuangg!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by PrashantMangukiya** [comment]
Waiting for confirmation from @sobitneupane

**Comment by puneetlath** [comment]
Just catching up here, but yes it is intentional not to show the list prior to searching. So I think we should do two things: 

1. Show some placeholder text when the textinput is empty (e.g. "Search to see options")
2. Show some text when the textInput is not empty but there are no results (e.g. "No results found")

**Comment by PrashantMangukiya** [comment]
@jasperhuangg It looks like @sobitneupane is OOO. Should I wait for him to be available or submit pr. I have to make sure it will not affect pr merge timeline bonus. Thanks.

**Comment by sobitneupane** [comment]
@PrashantMangukiya You can create PR. Once you are assigned to the issue, you are all good to create PR.

**Comment by PrashantMangukiya** [comment]
ok thank you

**Comment by dhairyasenjaliya** [comment]
@sobitneupane bump on comment https://github.com/Expensify/App/issues/15779#issuecomment-1484575194

**Comment by PrashantMangukiya** [comment]
@sobitneupane  PR is ready for review.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.91-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16602


If no regressions arise, payment will be issued on 2023-04-05. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@sobitneupane / @jasperhuangg] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@sobitneupane / @jasperhuangg] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@sobitneupane / @jasperhuangg] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@muttmuure] Determine if we should create a regression test for this bug.
- [ ] [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@muttmuure] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by PrashantMangukiya** [comment]
Ping for Upwork

**Comment by PrashantMangukiya** [comment]
Ping for Upwork

**Comment by PrashantMangukiya** [comment]
Ping for UpWork.

**Comment by PrashantMangukiya** [comment]
Ping For Upwork.

**Comment by jasperhuangg** [comment]
@muttmuure Can we make sure @PrashantMangukiya gets paid out for this? Thanks!

**Comment by dhanashree-sawant** [comment]
Hi @jasperhuangg, @muttmuure, also me for reporting bonus

**Comment by muttmuure** [comment]
Paying out today

**Comment by muttmuure** [comment]
Invited @dhanashree-sawant and @sobitneupane as C+

**Comment by muttmuure** [comment]
Contract sent to @PrashantMangukiya

**Comment by PrashantMangukiya** [comment]
@muttmuure Offer accepted on upwork.

**Comment by PrashantMangukiya** [comment]
Received Thank you.

**Comment by muttmuure** [comment]
Still waiting on @sobitneupane to accept offer

**Comment by sobitneupane** [comment]
@muttmuure I didn't review the proposal and PR. So, no payment for me. Apologies for responding late.

**Comment by muttmuure** [comment]
ok cool, thanks!
